packages/client/src/__tests__/integration/happy/count/test.ts

```diff
@@ -1,68 +1,165 @@
 import { getTestClient } from '../../../../utils/getTestClient'
 
-test('count', async () => {
-  const PrismaClient = await getTestClient()
-  const prisma = new PrismaClient()
-  const result = await prisma.user.count()
-  expect.assertions(4)
-  expect(result).toMatchInlineSnapshot(`10`)
-
-  const result2 = await prisma.user.count({
-    select: true,
+let PrismaClient, prisma
+
+describe('count', () => {
+  beforeAll(async () => {
+    PrismaClient = await getTestClient()
   })
-  expect(result2).toMatchInlineSnapshot(`10`)
-
-  const result3 = await prisma.user.count({
-    select: {
-      _all: true,
-      email: true,
-      age: true,
-      name: true,
-    },
+
+  beforeEach(() => {
+    prisma = new PrismaClient()
   })
-  expect(result3).toMatchInlineSnapshot(`
-    Object {
-      _all: 10,
-      age: 10,
-      email: 10,
-      name: 10,
-    }
-  `)
-  try {
-    await prisma.user.count({
+
+  afterEach(() => {
+    prisma.$disconnect()
+  })
+
+  test('simple', async () => {
+    const value = await prisma.user.count()
+
+    expect(value).toMatchInlineSnapshot(`10`)
+  })
+
+  test('where', async () => {
+    const value = await prisma.user.count({
+      where: {
+        age: 84,
+      },
+    })
+
+    expect(value).toMatchInlineSnapshot(`1`)
+  })
+
+  test('select where', async () => {
+    const value = await prisma.user.count({
+      select: true,
+      where: {
+        age: 84,
+      },
+    })
+
+    expect(value).toMatchInlineSnapshot(`1`)
+  })
+
+  test('select mixed where', async () => {
+    const value = await prisma.user.count({
       select: {
         _all: true,
         email: true,
         age: true,
         name: true,
-        posts: true,
+      },
+      where: {
+        age: 84,
       },
     })
-  } catch (err) {
-    expect(err.message).toMatchInlineSnapshot(`
 
-      Invalid \`prisma.user.count()\` invocation:
+    expect(value).toMatchInlineSnapshot(`
+      Object {
+        _all: 1,
+        age: 1,
+        email: 1,
+        name: 1,
+      }
+    `)
+  })
 
-      {
-        select: {
-          _count: {
-            select: {
-      ?       _all?: true,
-      ?       email?: true,
-      ?       age?: true,
-      ?       name?: true,
-              posts: true,
-              ~~~~~
-      ?       id?: true
+  test('select all true', async () => {
+    const value = await prisma.user.count({
+      select: true,
+    })
+
+    expect(value).toMatchInlineSnapshot(`10`)
+  })
+
+  // TODO: this should happen but we always make selects to become true
```

Comment:
We should create an issue for this 👍

Reply:
Definitely. The reason why I chose not to fix this is that it's been behaving like this before this regression even.

Reply:
yep, that would definitely be out of scope

Reply:
Actually, thinking about it, I don't think it's a problem. If a user does not want to make a selection with `select: false` that still means that he wants to count (but without any selection), hence `select: true` in any case.

Reply:
(it's confusing from a code logic point of view, but makes sense from a user point of view)

Reply:
It would be nice to have a test case with `select all false` that is passing so it changes the test will catch it right away.
With a comment for the test